Ginger/GingerCoreNET/Drivers/CoreDrivers/Web/Playwright/PlaywrightBrowserTab.cs

```diff
@@ -1048,6 +1048,167 @@ public async Task StartListenDialogsAsync()
             isDialogDismiss = false;
         }
 
+        /// <summary>
+        /// Waits for the URL to match the specified pattern within the given timeout.
+        /// </summary>
+        public async Task<bool> WaitForUrlMatchAsync(string urlPattern, float timeout)
+        {
+            ThrowIfClosed();
+            try
+            {
+                await _playwrightPage.WaitForURLAsync(urlPattern, new PageWaitForURLOptions { Timeout = timeout });
+                return true;
+            }
+            catch (Exception ex)
+            {
```

Comment:
it should return false in case URL match fails, for other kind exceptions, it should throw the exceptions